dht/token.cc

```diff
@@ -29,8 +29,8 @@
 
 namespace dht {
 
-static const token min_token{ token::kind::before_all_keys, {} };
-static const token max_token{ token::kind::after_all_keys, {} };
+static const token min_token{ token::kind::before_all_keys, std::array<uint8_t, 8>{} };
+static const token max_token{ token::kind::after_all_keys, std::array<uint8_t, 8>{} };
 
```

Comment:
Unrelated note: I'd like to remove token::kind too, and replace it with nonwrapping_range's support for open-ended ranges.

Reply:
I had exactly the same thought. I will work on this in the following PR.